fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java

```diff
@@ -0,0 +1,300 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.stats;
+
+import org.apache.doris.nereids.stats.FilterEstimation.EstimationContext;
+import org.apache.doris.nereids.trees.expressions.And;
+import org.apache.doris.nereids.trees.expressions.ComparisonPredicate;
+import org.apache.doris.nereids.trees.expressions.CompoundPredicate;
+import org.apache.doris.nereids.trees.expressions.EqualTo;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.GreaterThan;
+import org.apache.doris.nereids.trees.expressions.GreaterThanEqual;
+import org.apache.doris.nereids.trees.expressions.InPredicate;
+import org.apache.doris.nereids.trees.expressions.LessThan;
+import org.apache.doris.nereids.trees.expressions.LessThanEqual;
+import org.apache.doris.nereids.trees.expressions.Not;
+import org.apache.doris.nereids.trees.expressions.Or;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.nereids.trees.expressions.SlotReference;
+import org.apache.doris.nereids.trees.expressions.literal.Literal;
+import org.apache.doris.nereids.trees.expressions.visitor.ExpressionVisitor;
+import org.apache.doris.statistics.ColumnStat;
+import org.apache.doris.statistics.StatsDeriveResult;
+
+import com.google.common.base.Preconditions;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Calculate selectivity of expression that produces boolean value.
+ * TODO: Should consider the distribution of data.
+ */
+public class FilterEstimation extends ExpressionVisitor<StatsDeriveResult, EstimationContext> {
+
+    private static final double DEFAULT_SELECTIVITY = 0.1;
+
+    private static final double DEFAULT_INEQUALITY_COMPARISON_SELECTIVITY = 1.0 / 3.0;
+
+    private static final double DEFAULT_EQUALITY_COMPARISON_SELECTIVITY = 0.1;
+
+    private final StatsDeriveResult stats;
+
+    public FilterEstimation(StatsDeriveResult stats) {
+        Preconditions.checkNotNull(stats);
+        this.stats = stats;
+    }
+
+    /**
+     * This method will update the stats according to the selectivity.
+     */
+    public StatsDeriveResult estimate(Expression expression) {
+        // For a comparison predicate, only when it's left side is a slot and right side is a literal, we would
+        // consider is a valid predicate.
+        StatsDeriveResult stats = calculate(expression);
+        double expectedRowCount = stats.getRowCount();
+        for (ColumnStat columnStat : stats.getSlotToColumnStats().values()) {
+            if (columnStat.getNdv() > expectedRowCount) {
```

Comment:
Here we missed a lot of useful information. For example, expression is `A=1`, A.stat.ndv should be 1.